eng/README.md

```diff
@@ -0,0 +1,36 @@
+# `Eng` directory
```

Comment:
FYI @mikeharder I added this to capture the conventions you mentioned on this PR review.